packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -131,6 +131,97 @@ enum BottomNavigationBarType {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=stateful_widget_material}
+/// This example shows [BottomNavigationBar] used in a [Scaffold] Widget with
+/// different interaction patterns. Tapping on the first [BottomNavigationBarItem]
+/// uses the [ScrollController] to animate the [ListView] to the top. The second
+/// [BottomNavigationBarItem] opens a [DropdownButton].
+///
+///```dart
+/// int _selectedIndex = 0;
+/// final ScrollController _homeController = ScrollController();
+/// String dropdownValue = 'One';
+/// static const TextStyle optionStyle = TextStyle(fontSize: 15, fontWeight: FontWeight.bold);
+///
+/// void _onItemTapped(int index) {
+///   if (index == 0) {
+///     _homeController.animateTo(
+///       0.0,
+///       duration: Duration(milliseconds: 500),
+///       curve: Curves.easeOut
+///     );
+///   }
+///   setState(() {
+///     _selectedIndex = index;
+///   });
```

Comment:
I think that having the item be highlighted when zooming back the top of the page or when the modal/menu button is tapped doesn't quite make sense here. Maybe we should introduce new two items where tapping and selecting them actually make sense and causes changes on the UI. For example:

Button 1 scrolls to the top of the screen, but doesn't update the selected index since it doesn't make sense to do that when this button is tapped
Button 2 can become selected, and causes some kind of change on the screen (e.g. bolding all text on the screen)
Button 3 can become selected, and causes some kind of change on the screen (e.g. making all text on the screen normal weight)

Reply:
^ And then, doing a separate, similar example for the modal/dropdown sample